include/circt/Dialect/FIRRTL/FIRRTLStatements.td

```diff
@@ -50,7 +50,7 @@ def ConnectOp : FIRRTLOp<"connect", [FConnectLike]> {
   let hasCanonicalizer = true;
 }
 
-def StrictConnectOp : FIRRTLOp<"strictconnect", [SameTypeOperands, FConnectLike]> {
```

Comment:
It would be worth writing the constraint here rather than customizing the verifier.  Then it can apply to RefDefineOp also.